pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -607,6 +609,44 @@ protected void handlePartitionMetadataRequest(CommandPartitionedTopicMetadata pa
             isTopicOperationAllowed(topicName, TopicOperation.LOOKUP, authenticationData, originalAuthData).thenApply(
                     isAuthorized -> {
                 if (isAuthorized) {
+                    // Get if exists, respond not found error if not exists.
+                    if (!partitionMetadata.isMetadataAutoCreationEnabled()) {
+                        final NamespaceResources namespaceResources = getBrokerService().pulsar().getPulsarResources()
+                                .getNamespaceResources();
+                        final TopicResources topicResources = getBrokerService().pulsar().getPulsarResources()
+                                .getTopicResources();
+                        return namespaceResources.getPartitionedTopicResources()
+                            .getPartitionedTopicMetadataAsync(topicName, false)
+                            .thenAccept(metadata -> {
+                                if (metadata.isPresent()) {
+                                    commandSender.sendPartitionMetadataResponse(metadata.get().partitions, requestId);
+                                    lookupSemaphore.release();
+                                    return;
+                                }
+                                if (topicName.isPersistent()) {
+                                    topicResources.persistentTopicExists(topicName).thenAccept(exists -> {
+                                        if (exists) {
+                                            commandSender.sendPartitionMetadataResponse(0, requestId);
+                                            lookupSemaphore.release();
+                                            return;
+                                        }
+                                        writeAndFlush(Commands.newPartitionMetadataResponse(ServerError.TopicNotFound,
+                                                        "", requestId));
+                                        lookupSemaphore.release();
+                                    }).exceptionally(ex -> {
+                                        log.error("{} {} Failed to get partition metadata", topicName,
+                                                ctx.channel(), ex);
+                                        writeAndFlush(
+                                                Commands.newPartitionMetadataResponse(ServerError.AuthorizationError,
+                                                "Failed to check partition metadata",
+                                                requestId));
+                                        lookupSemaphore.release();
+                                        return null;
+                                    });
+                                }
+                            });
```

Comment:
add .exceptionally() here to cover the exceptions from `.getPartitionedTopicMetadataAsync`

Reply:
Fixed. Thanks